src/login/pages/Login.tsx

```diff
@@ -149,7 +149,13 @@ export default function Login(
                                     </div>
                                 </div>
                             )}
-                            <h5>{msgStr("selfCredentials")}</h5>
+                            {(() => {
+                                if (social.providers.length) {
+                                    return (
+                                        <h5>{msgStr("selfCredentials")}</h5>
+                                    );
+                                }
+                            })()}
```

Comment:
```suggestion
                            {social.providers.length !== 0 && <h5>{msgStr("selfCredentials")}</h5>}                           
```

Same thing esentially but more compact.